hack/generator/pkg/jsonast/schema_abstraction_openapi.go

```diff
@@ -0,0 +1,288 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package jsonast
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/url"
+
+	"github.com/go-openapi/jsonpointer"
+	"github.com/go-openapi/spec"
+	"github.com/pkg/errors"
+)
+
+// OpenAPISchema implements the Schema abstraction for go-openapi
+type OpenAPISchema struct {
+	inner     spec.Schema
+	root      spec.Swagger
+	fileName  string
+	groupName string
+	version   string
+
+	cache *OpenAPISchemaCache
+}
+
+// OpenAPISchemaCache is a cache of schema that have been loaded,
+// identified by file path
+type OpenAPISchemaCache struct {
+	files map[string]spec.Swagger
+}
+
+// MakeOpenAPISchemaCache creates an OpenAPISchemaCache with the initial
+// file path → spec mapping
+func MakeOpenAPISchemaCache(specs map[string]spec.Swagger) *OpenAPISchemaCache {
```

Comment:
minor: Other places I think we had `NewXxx` returns a ptr, whereas `MakeXxx` returns a non-ptr.

Here `MakeXxx` is returning a ptr.

Reply:
Same comment on a few other functions in this file

Reply:
Done.

Reply:
Actually this can be a value type again now that there are not mutexes involved. I will rerefactor.

Reply:
Done ×2.